ingestion/tests/magen_ingestion_test.py

```diff
@@ -1073,6 +1073,53 @@ def test_GridFS_UploadStream(self):
             for filename in glob.glob(IngestionGlobals().data_dir + "/" + file_name + "*"):
                 os.remove(filename)
 
+    def test_GridFS_UploadPublicFileStream(self):
+        """
+        Creates a local file, encrypts with API then tries to decrypt with OpenSSL. Original file and decrypted
+        file are compared for equality.
+        """
+        print("+++++++++ Encrypt_Asset_Decrypt_With_OpenSSL Test +++++++++")
+
+        file_name = "test_up.pub"
+        base_path = type(self).ingestion_globals.data_dir
+        src_file_full_path = os.path.join(base_path, file_name)
+        grid_file_full_path = src_file_full_path + ".grid"
+        fs = gridfs.GridFSBucket(type(self).db.core_database.get_magen_mdb(), bucket_name="test")
+        iid = 0
+        try:
+            # Create file
+            magen_file = open(src_file_full_path, 'w+')
+            magen_file.write("this is a test")
+            magen_file.close()
+            # Upload file
+            magen_file_upload = open(src_file_full_path, 'rb')
+            iid = fs.upload_from_stream("unit_test_grid", magen_file_upload,
```

Comment:
In order for this test to be useful you need to use metadata type "public key"  and search the file based on username and type = public key. Otherwise it does not reflect the new code.